### PR TITLE
[WEB-2871,WEB-2872] Personal Signup fixes to redirects

### PR DESCRIPTION
**Pull request by clintonium-119**
[WEB-2871] [WEB-2872]
Fixes some incorrect redirects for personal signup, and fix so that data storage is not set up when creating a viewonly patient account, and also, the first and last name fields are filled by default when returning to the form when the users clicks the CTA to add data storage.

[WEB-2871]: https://tidepool.atlassian.net/browse/WEB-2871?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-2872]: https://tidepool.atlassian.net/browse/WEB-2872?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/redux/actions/async.js [hunk 1]
```diff
@@ -1168,6 +1167,8 @@ export function fetchPatientData(api, options, id) {
 
       dispatch(sync.fetchPatientDataSuccess(id));
 
+      console.log('handleFetchSuccess', location.pathname, fetchingPatientId, id)
+
```
Comment (krystophv) [inline app/redux/actions/async.js:1171]: Figuring this got left in accidentally
```suggestion
```

File: app/redux/actions/async.js [hunk 2]
```diff
@@ -764,7 +763,7 @@ export function updateSettings(api, patientId, settings) {
  * @param {userId} userId
  * @param  {Object} formValues
  */
-export function updateUser(api, formValues) {
+export function updateUser(api, formValues, options) {
```
Comment (krystophv) [inline app/redux/actions/async.js:766]: This new function param is never read, and no new calls seem to pass it in. Was it tacked on accidentally?
```suggestion
export function updateUser(api, formValues) {
```

**Review by krystophv — APPROVED** [review]
Have a couple comments/suggestions. Overall LGTM though 👍🏼

File: app/redux/actions/async.js (hunk 2 above)
Comment (clintonium-119) [reply app/redux/actions/async.js:766]: Ah yes, I'd been going about this a bit differently when I began this ticket, and accidentally left it in.

**Pull request merged**